lib/uklibid/include/uk/libid.h

```diff
@@ -6,4 +6,53 @@
 #ifndef LIBUKLIBID_H_
 #define LIBUKLIBID_H_
 
+#include <uk/config.h>
+#include <uk/arch/limits.h>
+#include <uk/bits/libid.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define UKLIBID_NONE __UI_MAX
+
+#define _uk_libid_static(l) \
+	__UKLIBID_ ## l
+
+/**
+ * Compile-time mapping of a library name to its unique library identifier
+ * WARNING: Any unknown library name (e.g., library not part of build) will
```

Comment:
Is it possible to have a user of this header that is not part of the build?

Reply:
Unikraft's build system should be restrictive enough that this should not be possible. This was a try to document assumptions that were taken during development just for the case we suddenly export headers from Unikraft to external builds.